### DISCUSSION TITLE
Package micropython project to a custom bin file

### DISCUSSION THREAD

**Discussion by djs0109** [micropython/micropython#9898] [Ideas]
Hi everyone,

I have developed a project with 8 program files and now want to make it easy to use, i.e. someone who want to use the project, do not need to copy all the files one by one. For this purpose, I would like to know, if there is a way to package all the program files to a single .bin file (like a custom firmware)? 

Since I do not have much experience before I work with micropython on ESP32, I appreciate any information :)

**Comment by Lobo-T** [comment]
Start here: https://docs.micropython.org/en/latest/reference/manifest.html
If you search for "freezing" on the old forum I'm sure a lot of relevant question and answers will show up.

**Reply by djs0109** [reply]
Thanks! That is exactly what I am seeking for

**Comment by djs0109** [comment]
For others who have the same problem, here is the [wiki](https://github.com/micropython/micropython/wiki/Building-Micropython-Binaries) that I find most useful
